include/hydrogen/device/gpu/cuda/CUDAManagement.hpp

```diff
@@ -2,13 +2,80 @@
 #define HYDROGEN_DEVICE_GPU_CUDAMANAGEMENT_HPP_
 
 #include <cuda_runtime.h>
+#include <cuComplex.h>
+#include <thrust/complex.h>
+
+namespace El
+{
+template <typename T>
+class Complex;
+}// namespace El
 
 namespace hydrogen
 {
 
 using gpuEvent_t = cudaEvent_t;
 using gpuStream_t = cudaStream_t;
 
+template <typename T>
+struct NativeGPUTypeT
+{
+    using type = T;
+};
+
+template <>
+struct NativeGPUTypeT<El::Complex<float>>
+{
+    using type = thrust::complex<float>;
+};
+
+template <>
+struct NativeGPUTypeT<El::Complex<double>>
+{
+    using type = thrust::complex<double>;
+};
+
+template <typename T>
+using NativeGPUType = typename NativeGPUTypeT<T>::type;
+
+template <typename T>
+struct GPUStaticStorageTypeT
```

Comment:
I also hate those names. The "static storage type" is just that -- a type to store bits that can be statically allocated. You cannot allocate `__shared__` blocks of `thrust::complex`, apparently, so this is a hack around that. Then the "native type" should support the usual basic arithmetic operations (`+`, `-`, `*=`, etc.). Any suggestions for better names? Maybe instead of "native type" call it `DeviceComputeType` or something? I'm at a loss.

Reply:
`NativeType` isn't perfect, but it's ok. It makes sense to me. I think documenting `GPUStaticStorageType` is the best call, even if the name isn't perfect, since its use-case is so narrow and subtle.

Reply:
Do the comments added in b9997b2 sufficiently address your concern?